extensions/extended-isupport.md

```diff
@@ -0,0 +1,69 @@
+---
+title: "`extended-isupport` Extension"
+layout: spec
+copyrights:
+  -
+    name: "Simon Ser"
+    period: "2024"
+    email: "contact@emersion.fr"
+---
+
+## Notes for implementing work-in-progress version
+
+This is a work-in-progress specification.
+
+Software implementing this work-in-progress specification MUST NOT use the
+unprefixed `extended-isupport` capability name. Instead, implementations SHOULD
+use the `draft/extended-isupportt` capability name to be interoperable with
+other software implementing a compatible work-in-progress version. The final
+version of the specification will use unprefixed capability names.
+
+## Introduction
+
+`RPL_ISUPPORT` is used to advertise metadata about the server as key-value
+entries. However, `RPL_ISUPPORT` is only sent by servers after connection
+registration. This undermines the usefulness of `RPL_ISUPPORT`: some of the
+metadata would be useful to clients prior to connection registration. This
+extension adds a mechanism to send `RPL_ISUPPORT` messages before connection
+registration, as well as an end of `RPL_ISUPPORT` list indication.
+
+## Implementation
+
+### `draft/extended-isupport` capability
+
+When the `draft/extended-isupport` capability is enabled by the client, the
+server MUST send one or more `RPL_ISUPPORT` messages followed by a
+`RPL_ENDOFISUPPORT` message. While `draft/extended-isupport` is enabled, the
```

Comment:
What happens if the client enables, disables, then re-enables the capability? Should we require servers to re-send the list then?

What happens if the client enables the capability after connection registration? Same question: should we require servers to re-send the list then?

The current approach tries to keep the spec as simple as possible (always require to re-send the list) by avoiding special cases.

Reply:
Tried to make all of this stuff more explicit:

> If the client enables `draft/extended-isupport` after connection registration,
the server MUST send one or more `RPL_ISUPPORT` messages. If the client
enables, disables and then enables again `draft/extended-isupport` at any time,
the server MUST send one or more `RPL_ISUPPORT` messages. If the client
requests the `draft/extended-isupport` capability while it's already enabled,
the server MUST NOT send any `RPL_ISUPPORT` messages in response.

Reply:
The edge-triggered behavior has been removed.